kafka-connect-s3/src/test/java/io/confluent/connect/s3/storage/S3OutputStreamTest.java

```diff
@@ -53,4 +67,77 @@ public void testPropagateOtherRetriableS3Exceptions() {
     when(s3Mock.initiateMultipartUpload(any())).thenThrow(new AmazonClientException(S3_EXCEPTION_MESSAGE));
     assertThrows(IOException.class, () -> stream.commit());
   }
+
+  @Test
+  public void testNewMultipartUploadAESSSE() throws IOException {
+
+    Map<String, String> props = createProps();
+    props.put(S3SinkConnectorConfig.SSEA_CONFIG, SSEAlgorithm.AES256.toString());
+    stream = new S3OutputStream(S3_TEST_KEY_NAME, new S3SinkConnectorConfig(props), s3Mock);
+
+    when(s3Mock.initiateMultipartUpload(any())).thenReturn(mock(InitiateMultipartUploadResult.class));
+    stream.newMultipartUpload();
+
+    verify(s3Mock).initiateMultipartUpload(captor.capture());
+
+    assertNotNull(captor.getValue());
+    assertNotNull(captor.getValue().getObjectMetadata());
+    assertNull(captor.getValue().getSSECustomerKey());
+    assertNull(captor.getValue().getSSEAwsKeyManagementParams());
+  }
+
+  @Test
+  public void testNewMultipartUploadKMSSSE() throws IOException {
+
+    Map<String, String> props = createProps();
+    props.put(S3SinkConnectorConfig.SSEA_CONFIG, SSEAlgorithm.KMS.toString());
+    props.put(S3SinkConnectorConfig.SSE_KMS_KEY_ID_CONFIG, "key1");
+    stream = new S3OutputStream(S3_TEST_KEY_NAME, new S3SinkConnectorConfig(props), s3Mock);
```

Comment:
Is KMS without key a valid case? If yes, we should add a test for it as well.

Reply:
KMS algorithm will need an associated KMS Id to work, other wise it will result in error.
But, possibly a good validation to include